### ISSUE TITLE
Creating Collector pod with labelmap action fails when TargetAllocator is enabled

### ISSUE THREAD

**Issue by anunarapureddy** [open-telemetry/opentelemetry-operator#1623]
Describe the bug
Failed to create collector pod with the below error log:

```
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading receivers configuration for "prometheus": prometheus receiver failed to
 unmarshal yaml to prometheus config: "" is invalid 'replacement' for labelmap `action`
```

This is happening when loading a valid prometheus configuration into the opentelemetrycollector resource.

Input prometheus configuration:

```
relabel_configs:
- regex: (.*id)
  action: labelmap
```

The operator creates a configmap using the input prometheus configuration where the labelmap action is transformed into the below configuration:

```
relabel_configs:
- separator: ;
  regex: (.*id)
  replacement: $1
  action: labelmap
```

At a glance,`opentelemetry-operator`'s yaml marshal/unmarshall logic seems to be adding two extra fields "replacement" and "separator" while building the configmap. The input config is being rewritten using `DefaultRelabelConfig` which seems to be appending extra defaults.  The `DefaultRelabelConfig` lives in [prometheus codebase](https://github.com/prometheus/prometheus/blob/3923e83413075a17c5a10739696a879765a05542/model/relabel/relabel.go#L103) and is being referred by the [operator code](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/collector/reconcile/config_replace.go#L51-L62) to build the configmap only when `TargetAllocator` is enabled.

<img width="454" alt="Screen Shot 2023-04-03 at 8 43 07 PM" src="https://user-images.githubusercontent.com/23454368/229681742-7d5ee713-edc9-465a-9472-70665bc78d3a.png">

Steps to reproduce:

1. Create an opentelemetrycollector in statefulset mode with targetAllocator enabled.
2. Configure the prometheus receiver by providing a valid scrape configuration which includes `labelmap` action under `relabel_configs` section.
3. Wait for the creation of the collector pod (fails with error which results in the pod going into a CrashLoopBackoff)
4. Describe the configmap generated by the operator (labelmap with two extra fields appended)

What did you expect to see?

I expected to see the opentelemetry-operator create a collector pod successfully.

What did you see instead?
Pod creation fails with the below error which results in the pod going into a CrashLoopBackoff

```
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading receivers configuration for "prometheus": prometheus receiver failed to
 unmarshal yaml to prometheus config: "" is invalid 'replacement' for labelmap `action`
```

What configuration did you use?

Prometheus scrape configuration:
```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: metric-gateway-collectors
spec:
  mode: statefulset
  targetAllocator:
    enabled: true
    image: ghcr.io/open-telemetry/opentelemetry-operator/target-allocator:0.74.0
    replicas: 2
    allocationStrategy: consistent-hashing
  config: |
    exporters:
      logging:
        loglevel: info
    receivers:
      prometheus:
        config:       
          scrape_configs:          
          - job_name: kubernetes-nodes
            honor_timestamps: true
            scrape_interval: 1m
            scrape_timeout: 10s
            metrics_path: /metrics
            scheme: https
            authorization:
              type: Bearer
              credentials_file: /var/run/secrets/kubernetes.io/serviceaccount/token
            tls_config:
              ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
              insecure_skip_verify: true
            relabel_configs:
            - separator: ;
              regex: __meta_kubernetes_node_label_(.+)
              replacement: $1
              action: labelmap
            kubernetes_sd_configs:
            - role: node
              kubeconfig_file: ""
              follow_redirects: true   
        target_allocator:
          endpoint: http://metric-gateway-collectors-targetallocator:80
          interval: 30s
          collector_id: ${POD_NAME}
          http_sd_config:
            refresh_interval: 60s
    service:
      pipelines:
        metrics:
          exporters:
          - logging
          processors: []
          receivers:
          - prometheus
  resources:
    limits:
      cpu: 1
      memory: 10Gi
    requests:
      cpu: 500m
      memory: 10Gi
  autoscaler:
    behavior:
      scaleUp:
        stabilizationWindowSeconds: 30
    maxReplicas: 6
    minReplicas: 3
    targetCPUUtilization: 80
    targetMemoryUtilization: 80
```

What version did you use?
opentelemetry helm charts (0.25.1), otel-operator - 0.74.0, otel-collector - 0.74.0, target-allocator-0.74.0

**Comment by anunarapureddy** [comment]
It works as expected only when TA is disabled and explicitly setting `replacement: $$1` in prometheus relabel_configs. Even with TA disabled, if we do not explicitly set `replacement: $$1` on `labelmap`, `labeldrop`, `labelkeep` actions, it defaults to `replacement: $1` and the collector pod fails to get created. replacement is an optional field on these actions and will default to `$1` if not explicitly set.

**Comment by anunarapureddy** [comment]
cc @open-telemetry/operator-ta-maintainers

**Comment by jaronoff97** [comment]
This should have been closed on merge of #1712 :)
